### PR TITLE
Update l3gd20 sensor driver with new sensor driver model

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
Support l3gd20 sensor with new sensor model on stm32f429i-discovery board.
## Impact
We can test l3gd20 gyroscope sensor on stm32f429i-discovery board.
## Testing
daily test

### CODE REVIEW

**Comment by v01d** [comment]
Can you remove the check for HPWORK define in code to Kconfig, only for when BUFFER > 0?

**Comment by v01d** [comment]
Also, why the need to introduce a config variable for the gyro which is actually on-board? I would presume this should always be zero (many boards do this).

**Comment by Donny9** [comment]
> Also, why the need to introduce a config variable for the gyro which is actually on-board? I would presume this should always be zero (many boards do this).

okay, i revert this commit

**Comment by Donny9** [comment]
> Can you move the check for HPWORK define in code to Kconfig, only for when BUFFER > 0?

done

**Comment by xiaoxiang781216** [comment]
@v01d is the PR ready for merge?

**Comment by v01d** [comment]
Actually I requested for the check for HPWORK to be moved to Kconfig, but maybe I wasn't clear. I imagine you can select HPWORK if buffer size > 0. This avoids trial and error building.

**Comment by Donny9** [comment]
> Actually I requested for the check for HPWORK to be moved to Kconfig, but maybe I wasn't clear. I imagine you can select HPWORK if buffer size > 0. This avoids trial and error building.

done

File: drivers/sensors/Kconfig [hunk 1]
```diff
@@ -222,9 +222,19 @@ config SENSORS_L3GD20
 	bool "STMicro L3GD20 Gyroscope Sensor support"
 	default n
 	select SPI
+	select SCHED_HPWORK if SENSORS_L3GD20_BUFFER_SIZE > 0
```
Comment (xiaoxiang781216) [inline drivers/sensors/Kconfig:225]: should we move to SENSORS_L3GD20_BUFFER_SIZE?

**Comment by v01d** [comment]
I think the placement is OK

**Review by v01d — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
> I think the placement is OK
> EDIT: I was answering @xiaoxiang781216 comment which I received by email, but don't know why it does not appear here

I talk with @Donny9 privately and he has tried put the select in SENSORS_L3GD20_BUFFER_SIZE, but kconfig-frontend don't accept that if expression contain self.

**Pull request merged**